posts/inside-rust/test-infra-jan-feb-2025.md

```diff
@@ -0,0 +1,182 @@
+
+The migration effort took around a year, until we were finally able to declare all `Makefile`-based `run-make` tests ported, and thus we were able to [retire the legacy `Makefile`-based test infrastructure in early 2025][retirement].
+
+*Of course*, the new test infrastructure isn't all sunshine and rainbows. There are still issues, desired improvements and test UX papercuts that await to be addressed. However, like the overall test infra, they can be and will be improved over time.
+
+[^grep]: The test suite even had to maintain behavioral tests for `grep` because there are *different* flavors of `grep` that are incompatible with each other and had different CLI interfaces / behavior.
+[^broken]: During the porting process, we found multiple tests that had varying degree of brokenness due to hard to notice `Makefile` and shell quirks.
+
+[run-make-v2]: https://github.com/rust-lang/rust/pull/113026
+[`object`]: https://crates.io/crates/object
+[`regex`]: https://crates.io/crates/regex
+[run-make-port]: https://github.com/rust-lang/rust/issues/121876
+[`try-job`]: https://rustc-dev-guide.rust-lang.org/tests/ci.html#try-builds
+[gsoc-2024]: https://summerofcode.withgoogle.com/archive/2024/projects/P5BC91Hr
+[Oneirical]: https://github.com/oneirical
+[gsoc-final]: https://oneirical.github.io/gsocfinal/
+[retirement]: https://github.com/rust-lang/rust/pull/136581
+
+### `ci.py` is now a proper `citool` Rust crate
+
+The old `ci.py` Python script used to orchestrate CI jobs was unmaintainable. Any changes to the python script risked bringing down the entire queue or bypass testing entirely. There was practically no test coverage. CI UX improvements were hard to implement and difficult to review.
+
+So, Jakub decided enough was enough and [rewritten `src/ci/github-actions/ci.py` as `src/ci/citool`](https://github.com/rust-lang/rust/pull/136864), a proper Rust CLI tool. This allowed the job definitions to be properly parsed and handled, and also enabled adding unit tests. It also allowed improving some error messages. Furthermore, it improved the UX of running the CI jobs locally (on Linux). Consult the [`rustc-dev-guide` docs in `rust-lang/rust`](https://github.com/rust-lang/rust/blob/master/src/doc/rustc-dev-guide/src/tests/ci.md#docker) for updated running instructions (at the time of writing, this hasn't been synced back to [rust-lang/rustc-dev-guide] yet).
+
+### Bootstrap test and build step metrics are now available in GitHub job summaries
+
+<https://github.com/rust-lang/rust/pull/137077> implemented postprocessing logic for bootstrap test and build metrics to convert them into [GitHub job summaries][github-job-summaries].
+
+![Sample job summary](../../../../images/2025-03-11-test-infra-jan-feb-2025/example-ci-job-summary.png)
+
+[github-job-summaries]: https://github.blog/news-insights/product-news/supercharging-github-actions-with-job-summaries/
+
+
+## Notable changes
+
```

Comment:
I think https://github.com/rust-lang/rust/pull/137215 is definitely worth to include at this part.

Reply:
Ah yeah. Good point.

Reply:
Added a note for that.